helix-term/src/ui/picker.rs

```diff
@@ -226,7 +228,8 @@ impl<T: Item + 'static> Component for FilePicker<T> {
                 })
                 .unwrap_or(0);
 
-            let offset = Position::new(first_line, 0);
+            preview_scroll_offset = preview_scroll_offset.min(doc.text().len_lines());
```

Comment:
Rather than in the render loop, shouldn't we do this in `move_preview_by`?

Reply:
Meaning having FilePicker::move_preview_by() instead of Picker::move_preview_by()?

Reply:
`move_preview_by()` is implemented in Picker and I needed the `len_lines()` of the current file in order to limit the scroll. I could transfer that functionality to a `FilePicker::move_preview_by()` function and handle the preview scroll events in `FilePicker::handle_event()`